pkg/apis/serving/metadata_validation.go

```diff
@@ -33,7 +33,7 @@ import (
 
 var (
 	allowedAnnotations = map[string]struct{}{
-		CheckValidityOnDeploy:            {},
+		CheckValidityOnDeployAnnotation:  {},
```

Comment:
Should it be `CheckReadiness`?

Reply:
@vagababov we've been going back and forth on the naming. I'm trying to get the functionality right first and we can decide on the final name when we socialize this feature with the wg.